.teamcity/builds/Build.kt

```diff
@@ -15,33 +15,47 @@ class Build(
       this.id(name.toId())
       this.name = name
 
-      val packaging =
-          Maven("${name}-package", "package", "package", "-pl :packaging -am -DskipTests")
-
       val complete = Empty("${name}-complete", "complete")
 
       val bts = sequential {
         if (forPullRequests)
             buildType(WhiteListCheck("${name}-whitelist-check", "white-list check"))
         if (forPullRequests) dependentBuildType(PRCheck("${name}-pr-check", "pr check"))
-        dependentBuildType(Maven("${name}-build", "build", "test-compile"))
-        dependentBuildType(Maven("${name}-unit-tests", "unit tests", "test"))
-        dependentBuildType(collectArtifacts(packaging))
-        dependentBuildType(
-            IntegrationTests("${name}-integration-tests", "integration tests") {
-              dependencies {
-                artifacts(packaging) {
-                  artifactRules =
-                      """
+
+        parallel {
+          listOf("11", "17").forEach { javaVersion ->
+            val packaging =
+                Maven(
+                    "${name}-package",
+                    "package",
+                    "package",
+                    javaVersion,
+                    "-pl :packaging -am -DskipTests")
+
+            sequential {
+              dependentBuildType(Maven("${name}-build", "build", "test-compile", javaVersion))
+              dependentBuildType(Maven("${name}-unit-tests", "unit tests", "test", javaVersion))
+              dependentBuildType(collectArtifacts(packaging))
+              dependentBuildType(
+                  IntegrationTests("${name}-integration-tests", "integration tests", javaVersion) {
```

Comment:
I think one problem with the new setup is that the integration tests run on the desired java version, but we do not do anything about the java version on the Kafka Connect container - which is the actual thing, or am I missing something?

Reply:
Hmm, I see now that we also migrated the images to the latest version of Confluent Platform.